src/parser.js

```diff
@@ -239,12 +242,18 @@ export class Parser extends Tokenizer {
           directives.push(this.markLocation({ type: "Directive", rawValue }, directiveLocation));
         } else {
           parsingDirectives = false;
+          if (directiveOctal && this.strict) {
+            throw directiveOctal;
+          }
           statements.push(stmt);
         }
       } else {
         statements.push(stmt);
       }
     }
+    if (directiveOctal && this.strict) {
```

Comment:
This condition is only hit if there are no non-directive statements (in which L235 is not reached).

Strictly speaking L235 is redundant, but I think it's best to throw the error after parsing just the directives instead of waiting until we've parsed the entire body.